firebase-firestore/src/test/java/com/google/firebase/firestore/spec/SpecTestCase.java

```diff
@@ -241,7 +241,8 @@ public static void log(String line) {
   // Methods for tracking state of writes.
   //
 
-  abstract Persistence getPersistence(boolean garbageCollectionEnabled);
+  protected abstract ComponentProvider initializeComponentProvider(
+      ComponentProvider.Configuration configuration, final boolean garbageCollectionEnabled);
```

Comment:
Nit: I think Java recommendations discourage `final` on parameters and local variables.

Reply:
Removed. I added it as it is used in an inner class, but it is "effectively final" without it.